### DISCUSSION TITLE
405 Method not allowed from Next APIs

### DISCUSSION THREAD

**Discussion by alimoli** [vercel/next.js#69201] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/limoli/project-405

### To Reproduce

1. Execute `yarn dev`
2. Test against `localhost:3000`:
```
curl -X POST http://localhost:3000/api/hello/world  -I

Expected/Actual Output:
HTTP/1.1 200 OK
```
3. Deploy to Vercel (already available, see https://project-405.vercel.app)
4. Test against the Vercel deployment:
```
curl -X POST https://project-405-git-main-limoli.vercel.app/api/hello/world  -I

Expected Output:
HTTP/1.1 200 OK

❌ Actual Output:
HTTP/2 405
```

### Current vs. Expected behavior

The issue is caused by the presence of the `api` folder at the root of the project. 
This folder is not the real API folder containing the `route.ts` (which is under /app), but it is just a folder for utils that I wanted to use to include some hooks and related utilities.
Moving that `api` folder to a different folder at the same level (f.e. `utils`) solves the issue on the deployment.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.5.0: Wed May  1 20:14:38 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6020
  Available memory (MB): 16384
  Available CPU cores: 12
Binaries:
  Node: 20.11.1
  npm: 10.2.4
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.2.6 // Latest available version is detected (14.2.6).
  eslint-config-next: 14.2.4
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.5.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Developer Experience, Documentation, Pages Router, Runtime

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

**Comment by samcx** [comment, marked as answer]
@alimoli Thank you for submitting this!

This is a Vercel-related gotcha. You cannot have a root `/api` folder—it'll clash with Vercel automatically trying to create a Vercel Function. I would just re-name the folder if you want that directory at the root.
